### ISSUE TITLE
Entities classes should implement Parcelable for better performance

### ISSUE THREAD

**Issue by mgujare** [Yelp/yelp-android#44]
@tzuhanwu any thoughts on using below gradle dependency ?
'com.ryanharter.auto.value:auto-value-parcel:0.2.0'

more here.
http://ryanharter.com/blog/2016/03/22/autovalue/

**Comment by tzuhanwu** [comment]
Hello @mgujare, @jbrick and I discussed about Parcelable a while ago, adding Parcelable will improve the performance but remove the compatibility to the regular JAVA. We have tried to keep the SDK be shareable between JAVA and Android. However, I do feel like we are having more discussion around the packages that only works for Android, and the shareability makes it more difficult to expand the SDK. I think it's a good time to consider making this SDK optimized for Android. 

@mittonk, @jbrick, how do you feel about it?

**Comment by jbrick** [comment]
Seems reasonable to me, we never did end up advertising it as a Java SDK.

**Comment by mgujare** [comment]
Since @jbrick mentioned about not advertising this repo as java sdk i think changes may be needed in build.gradle in order to add android specific dependencies for this particular issue. eg: applying android plugin. Not sure if gradle can recognize 'annotationProcessor' and 'provided' dependencies as mentioned in link above without android plugin. Another likely change to move integration tests under src/main/test directory. Of course all this makes sense if yelp wants to advertise this primarily as android sdk..what say ?

**Comment by tzuhanwu** [comment]
Yes, a change to build.gradle for android plugin will be the first step of doing this. We have decided to move forward with this SDK to support Android specifically. And since the incompatibility with the previous version, a major version bump will be required. 

I'll start to draft up the change for the build.gradle and related files, if anyone would like to do the honor, feel free to create a pull-request.

**Comment by mgujare** [comment]
👍
